xarray/core/dataset.py

```diff
@@ -1404,12 +1406,16 @@ def isel(self, drop=False, **indexers):
         Dataset.sel
         DataArray.isel
         """
+
+        indexers = combine_pos_and_kw_args(indexers, indexers_kwargs, 'isel')
+        assert isinstance(drop, bool)
```

Comment:
This should either be dropped for now, or raise `TypeError`. `assert` isn't appropriate for external APIs.